tests/components/zeversolar/test_init.py

```diff
@@ -27,34 +26,21 @@ async def test_async_setup_entry_fails(
         pytest.raises(ConfigEntryNotReady),
     ):
         await init.async_setup_entry(hass, config_entry)
-
+        assert config_entry.state is ConfigEntryState.SETUP_RETRY
 
     with (
         patch("homeassistant.components.zeversolar.PLATFORMS", []),
         patch("zeversolar.ZeverSolarClient.get_data", return_value=zeversolar_data),
     ):
-        config_entry.add_to_hass(hass)
-
         result = await hass.config_entries.async_setup(config_entry.entry_id)
         await hass.async_block_till_done()
 
         assert result is True
         assert config_entry.state is ConfigEntryState.LOADED
 
-
-async def test_load_entry_fails(
-    hass: HomeAssistant, config_entry: MockConfigEntry
-) -> None:
-    """Test to start the integration when inverter is offline (e.g. at night)."""
-
     with (
-        patch("homeassistant.components.zeversolar.PLATFORMS", [Platform.SENSOR]),
-        patch("zeversolar.ZeverSolarClient.get_data", side_effect=ZeverSolarTimeout),
+        patch("homeassistant.components.zeversolar.PLATFORMS", []),
     ):
-        config_entry.add_to_hass(hass)
-
-        result = await hass.config_entries.async_setup(config_entry.entry_id)
+        result = await init.async_unload_entry(hass, config_entry)
```

Comment:
Again, don't use `init`, use `config_entries.async_unload` instead.
```suggestion
        result = await hass.config_entries.async_unload(config_entry.entry_id)
```